src/routes/donors.js

```diff
@@ -101,18 +101,54 @@ router.get('/sums', function(req, res, next) {
   })
 })
 
-router.put('/updatedonors/:id', (req, res) => {
-  console.log('hello world')
-  db.Donor.update(
-    {
-      email: req.body.email
-    },
-    {
-      where: {
-        id: req.params.id
-      }
-    }
-  ).then(result => res.json(result))
+router.get('/edit', (req, res) => {
+  db.PreferredContact.findAll({
+    attributes: ['id', 'description']
+  })
+    .then(result => {
+      res.json(result)
+    })
+    .catch(error => {
+      res.status(500).json(error)
+    })
+})
+
+router.put('/edit/:idNo', (req, res) => {
+  const { idNo: donorId } = req.params
+  const { remarks, preferredContact, dnc } = req.body
+  if (!remarks /* && !preferredContact && !('dnc' in req.body) */) {
```

Comment:
These commented parts, is it intentional? This line is supposed to early exit the process if the endpoint was called with nothing to update (i.e. never pass remarks, never pass preferredContact and never pass dnc).

Reply:
Oh, i misunderstood this line. I thought it was to check if they are falsy values.